From 5c3fe453e8aa6033030e1885b3ddb14867dffc06 Mon Sep 17 00:00:00 2001
From: Kevin Suo <suokunlong@126.com>
Date: Thu, 15 Jul 2021 12:56:43 +0200
Subject: [PATCH] restore compatibility with older popplers

with poppler 20.09:

/home/rene/LibreOffice/git/libreoffice-7-2/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx: In member function 'int
                 pdfi::PDFOutDev::parseFont(long long int, GfxFont*, GfxState*) const':
/home/rene/LibreOffice/git/libreoffice-7-2/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx:438:39: error: 'class
                 GfxFont' has no member named 'getNameWithoutSubsetTag'

https://www.google.com/search?q=getNameWithoutSubsetTag&oq=getNameWithoutSubsetTag&aqs=chrome..69i57.784j0j7&sourceid=chrome&ie=UTF-8 suggests it was added in 20.12

Change-Id: I4eacd2d740cb689ff9b3c6cab59376e01b1ba162
---
 sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx b/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx
index d8b73f621a09..bc313f76ecc4 100644
--- a/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx
+++ b/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx
@@ -432,10 +432,15 @@ int PDFOutDev::parseFont( long long nNewId, GfxFont* gfxFont, GfxState* state )
     FontAttributes aNewFont;
     int nSize = 0;
 
-#if POPPLER_CHECK_VERSION(0, 64, 0)
-    const
-#endif
+#if POPPLER_CHECK_VERSION(20, 12, 0)
     std::string familyName = gfxFont->getNameWithoutSubsetTag();
+#else
+    std::string familyName = gfxFont->getName()->toStr();
+    if (familyName.length() > 7 && familyName.at(6) == '+')
+    {
+        familyName = familyName.substr(7);
+    }
+#endif
     if( familyName != "" )
     {
         aNewFont.familyName.clear();
-- 
2.30.2

